Mathlib/Topology/Constructions.lean

```diff
@@ -1053,6 +1054,64 @@ lemma IsClosedEmbedding.sumElim {f : X → Z} {g : Y → Z}
   rw [IsClosedEmbedding.isClosedEmbedding_iff_continuous_injective_isClosedMap] at hf hg ⊢
   exact ⟨hf.1.sumElim hg.1, h, hf.2.2.sumElim hg.2.2⟩
 
+lemma inl_image_eq_preimage_elim {f : X → Z} {g : Y → Z} (S : Set Z) :
+    Sum.inl '' (f ⁻¹' S) = Sum.elim f g ⁻¹' S := by
```

Comment:
TODO: this is only true in general if S is contained in range f (which is true in the case I'm applying it). Same for the inr version.